packages/query-core/src/queriesObserver.ts

```diff
@@ -155,29 +155,14 @@ export class QueriesObserver extends Subscribable<QueriesObserverListener> {
         !matchedQueryHashes.includes(defaultedOptions.queryHash),
     )
 
-    const unmatchedObservers = prevObservers.filter(
-      (prevObserver) =>
-        !matchingObservers.some((match) => match.observer === prevObserver),
-    )
-
     const getObserver = (options: QueryObserverOptions): QueryObserver => {
       const defaultedOptions = this.client.defaultQueryOptions(options)
       const currentObserver = this.observersMap[defaultedOptions.queryHash!]
       return currentObserver ?? new QueryObserver(this.client, defaultedOptions)
     }
 
     const newOrReusedObservers: QueryObserverMatch[] = unmatchedQueries.map(
-      (options, index) => {
-        if (options.keepPreviousData) {
-          // return previous data from one of the observers that no longer match
-          const previouslyUsedObserver = unmatchedObservers[index]
-          if (previouslyUsedObserver !== undefined) {
-            return {
-              defaultedQueryOptions: options,
-              observer: previouslyUsedObserver,
-            }
-          }
-        }
+      (options) => {
```

Comment:
will `placeholderData => placeholderData` work with `useQueries` ? I think this was one of the open questions from the RFC.

Reply:
So from what i found it worked this way:
- find all queries in the array that have the same `queryKey` as in previous pass - keep those observers intact
- find all queries in the array that were `present` in previous pass, but are no longer there (by `queryKey`) - unsubscribe from those observers
- find all queries in the array that were `not present` in previous pass, but were added to the current array (by `queryKey`) - subscribe to those observers

What `keepPreviousData` did before is to `reuse` observer that should be removed (unsubscribed) for queries that were added.
Although this seems like a good idea and it actually works for queries that have the same data structure, i do not think we should keep this behavior and therefore removed it (you also mentioned it in the RFC).

Since we do not actually know what kind of data structure new queries would have, we have no way of matching them with previous queries. Current method `might work`, but we can also end up with query returning `object` from `placeholder` and then `array` from `queryFn`.

Reply:
yeah I think this is the right call 👍 

does it make sense to change the structure of `placeholderData` (at least on type level) then to `() => TQueryData` and _not_ pass `previousData` to it?

Also, it seems like we still have a lot of tests for the old keepPreviousData? I've also seen that they are flaky from time to time - are they still needed ?

Reply:
Removed tests and changed the `placeholderData` signature for `useQueries`.
When we remove `context` we can potentially move `UseQueryOptionsForUseQueries` to core to be shared across packages.

Reply:
Sorry for the necropost, but I just hit the issue where `useQueries` doesn't support `keepPreviousData` due to not receiving the previous data, and it's not clear from the migration guide what's the expected workaround when the data shape doesn't change 🤔